apps/system/test/unit/browser_test.js

```diff
@@ -81,4 +92,26 @@ suite('system/Browser', function() {
     var app = MockAppWindowHelper.mLatest;
     assert.equal(app.isPrivate, true);
   });
+
+  test('when private browsing is the default setting', function() {
+    MockNavigatorSettings.mTriggerObservers('browser.private.default',
+        {settingValue: true});
+
+    setTimeout(function () {
```

Comment:
Ok - so is this the problem? Instead of doing this, have you tried using sinon fake timers?